app/assets/javascripts/discourse/app/routes/login.js

```diff
@@ -35,6 +41,15 @@ export default class LoginRoute extends DiscourseRoute {
     controller.set("flashType", "");
     controller.set("flash", "");
 
+    if (this.login.isOnlyOneExternalLoginMethod) {
+      if (this.siteSettings.login_required) {
+        controller.set("autoRedirect", true);
+        controller.set("singleExternalLogin", this.login.singleExternalLogin);
```

Comment:
Is this being used in the controller? Doesn't look like it.

Reply:
It's used in the template